### ISSUE TITLE
Unable to connect with username and password

### ISSUE THREAD

**Issue by larrychu-ibm** [dperson/samba#47]
After creating a container with a user, I'm unable to connect to the share using my windows 7 host with that username/password.

$ docker run -it -p 139:139 -p 445:445 dperson/samba -u "foo;bar" -s "public;/tmp;yes;no;no;foo"
Added user foo.
smbd version 4.5.2-Debian started.
Copyright Andrew Tridgell and the Samba Team 1992-2016
STATUS=daemon 'smbd' finished starting up and ready to serve connections

I'm running docker 1.12.0 docker-toolbox on windows 7.

On the other hand, I have no problems connecting to the share from my windows 7 host when started with this command.
$ docker run -it -p 139:139 -p 445:445 dperson/samba -u "foo;bar" -s "public;/tmp;yes;no;yes"

The error I see from my windows 7 samba client is 
Logon failure: unknown user name or bad password

![image](https://cloud.githubusercontent.com/assets/12495945/24114220/d1c52226-0d75-11e7-8aaf-9b15335f97c6.png)

**Comment by dperson** [comment]
I don't have a windows 7 system to test with, but can connect from both Linux (Fedora 25) and OSX (10.12.3), using the exact command line you provided as not working using the `foo` user and `bar` password to authenticate. I'm not sure that it's a samba or my configuration issue. Can you test from another device?

**Comment by larrychu-ibm** [comment]
Our team has the same setup each with a windows 7 machine with possibly various different versions of docker installed.  Our containers had been working until recently. Ever since we've updated the container to the latest image, they have stopped working. 

We also have other team members with mac OSX with native docker. The dperson/samba container running in mac OSX native docker works just fine.

As requested, I tested the same container with docker ubuntu cifs mount, and authentication works just fine. The only configuration I've found so far that I can't get past username/password authentication is when connecting from the windows 7 host running the docker virtual machine boot2docker.

**Comment by larrychu-ibm** [comment]
I hacked the samba.sh script and turned up the log level. This is some of what I see.

UNIX token of user 0
Primary group is 0 and contains 0 supplementary groups
account_policy_get: name: password history, val: 0
pop_sec_ctx (0, 0) - sec_ctx_stack_ndx = 1
Storing account foo with RID 1000
check lock order 1 for /var/lib/samba/private/passdb.tdb
lock order:  1:/var/lib/samba/private/passdb.tdb 2:<none> 3:<none>
Locking key 555345525F666F6F00
Allocated locked data 0x0x555c985d4f60
Unlocking key 555345525F666F6F00
release lock order 1 for /var/lib/samba/private/passdb.tdb
lock order:  1:<none> 2:<none> 3:<none>
tdb_update_sam: Updating key for RID 1000
check lock order 1 for /var/lib/samba/private/passdb.tdb
lock order:  1:/var/lib/samba/private/passdb.tdb 2:<none> 3:<none>
Locking key 5249445F303030303033
Allocated locked data 0x0x555c985fd5c0
Unlocking key 5249445F303030303033
release lock order 1 for /var/lib/samba/private/passdb.tdb
lock order:  1:<none> 2:<none> 3:<none>
pop_sec_ctx (0, 0) - sec_ctx_stack_ndx = 0
check_ntlm_password: sam authentication for user [foo] FAILED with error NT_STATUS_WRONG_PASSWORD
check_ntlm_password:  Authentication for user [foo] -> [foo] FAILED with error NT_STATUS_WRONG_PASSWORD
Checking NTLMSSP password for LARRY-THINKPAD\foo failed: NT_STATUS_WRONG_PASSWORD
../auth/ntlmssp/ntlmssp_server.c:737: Checking NTLMSSP password for LARRY-THINKPAD\foo failed: NT_STATUS_WRONG_PASSWORD
SPNEGO login failed: NT_STATUS_WRONG_PASSWORD

**Comment by dperson** [comment]
Are you running this on docker for windows? or is windows just the client?

I don't work with windows much, but looking at your login box it looks like you're trying to login as `dockerhost\foo`. AFAIUI (As Far As I Understand It) that means you're trying to authenticate to the `dockerhost` domain, which isn't configured with samba. Try clicking on the `Use another account` button and just use the `foo` username.

**Comment by larrychu-ibm** [comment]
Thanks for continuing to look at this. I've tried signing in with just foo, with dockerhost\foo, with <ip address>\foo. None of the above works. When signing in with just foo with windows 7, it automatically sets the machine name as the domain, LARRY-THINKPAD\foo. The log above is from signing in with just "foo" where windows 7 automatically assigns the localmachine host name as the domain name.

I'm running windows 7 on the host machine, LARRY-THINKPAD where I have installed docker which runs as the VirtualBox boot2docker VM. dockerhost is a host name in the windows 7 hosts file which maps to the default docker ip address which is 192.168.99.100.

**Comment by dperson** [comment]
Hmm, try mounting it from the command line (with `cmd.exe`):
```
net use q: \\dockerhost\public /USER:foo
```

**Comment by larrychu-ibm** [comment]
Thanks for that tip. I tried doing that, and got the same error.

However, I believe I have figured out that there is some failure in samba 4.5.x to properly negotiate the protocol for NTLMv2 with windows 7.

I changed the local security policy on my windows host machine (samba client) to "Send NTLMv2 response only. Refuse LM & NTLM" from the default setting which is "Send LM & NTLM - use NTLMv2 session security if negotiated"

This post led me to this conclusion.
https://social.technet.microsoft.com/Forums/windows/en-US/8249ad4c-69aa-41ba-8863-8ecd7a7a4d27/samba-share-password-refused?forum=win10itpronetworking

I'll close this issue, as it is a samba issue, and not a problem with the docker image.
